### PR TITLE
rg-5: init-database

### PR DESCRIPTION
**Pull request by Icerock10**
- Created scheme for database
- Updated `mermaid` diagram
- Added `migrations` with `relations`
- Added a few `models` with `relationMappings`
- Added `seeds`

### CODE REVIEW

**Review by apofis96 — CHANGES_REQUESTED** [review]
Use guid for ID

**Comment by Vitaliy-Kalenichenko** [comment]
I would change the database structure a little:
1. Add `UserProfile` table (firstName, lastName, avatar, etc...)
2. The `Images` table is redundant
3. In the `Users` table there is a “recovery code” field. I'm not sure we'll need it
4. `Templates` table - change the relation to users to many-to-many
5. `Resumes` table - not sure if a connection with the images table is needed
6. Consider adding fields for images to `Resumes` and `Templates` tables

File: readme.md [hunk 1]
```diff
@@ -41,21 +43,131 @@ TBA
...
+    personal_information |o--|| resumes : resume_id
+    reviews ||--|| resumes : resume_id
+    resumes ||--|| templates : resume_id
+    recently_viewed ||--|| users: user_id
+    recently_viewed ||--|| resumes: resume_id
+    recently_viewed ||--|| templates: template_id
+    users ||--o{ user_templates : "user_id"
+    templates ||--o{ user_templates : "template_id"
+    user_templates }o--|| users : "BELONGS TO"
+    user_templates }o--|| templates : "BELONGS TO"
```
Comment (apofis96) [inline readme.md:170]: Remove BELONGS TO from mermaid

**Review by apofis96 — CHANGES_REQUESTED** [review]

File: backend/src/migrations/20240210124346_create_tables.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            .dateTime(DatabaseColumnName.UPDATED_AT)
+            .notNullable()
+            .defaultTo(knex.fn.now());
+    });
+    await knex.schema.createTable(
+        DatabaseTableName.TECHNICAL_SKILLS,
+        (table) => {
+            table.uuid(DatabaseColumnName.ID).primary();
+            table.string(DatabaseColumnName.SKILL_NAME).notNullable();
+            table.string(DatabaseColumnName.SKILL_LEVEL).notNullable();
```
Comment (Vitaliy-Kalenichenko) [inline backend/src/migrations/20240210124346_create_tables.ts:74]: Consider using enum instead of string here

Comment (Icerock10) [reply backend/src/migrations/20240210124346_create_tables.ts]: Sure, good point)

File: readme.md (hunk 1 above)
Comment (Icerock10) [reply readme.md]: +

**Review by Vitaliy-Kalenichenko — APPROVED** [review]

**Review by apofis96 — APPROVED** [review]

**Pull request merged**